### DISCUSSION TITLE
how to operate on any path?

### DISCUSSION THREAD

**Discussion by Dieken** [lihaoyi/os-lib#312] [Q&A] [answered]
I need to read lines from a file path:

Java:

```
import java.nio.file.*

Paths.readAllLines(Paths.get(s))
```

Scala better-files:
```
import better.files.*

s.toFile.lines
```

Scala os-lib:
```
import os.*

read.lines(if FilePath(s).toNIO.isAbsolute then Path(s) else pwd / s)
```

The os-lib version is ugly, is there any better way to operate on any path? I would like something like `read.lines(s)`  or `read.lines(FilePath(s))`.

**Comment by lefou** [comment, marked as answer]
```scala
os.read.lines(os.Path(s, os.pwd))
```

**Reply by Dieken** [reply]
Thanks! Sorry I didn't notice the Readme.adoc already mentioned this usage, I read the scaladoc at https://javadoc.io/doc/com.lihaoyi/os-lib_3/latest/index.html, it basically has no document, I also asked some friends and LLM models, all failed to answer me :-(
